### ISSUE TITLE
Don't automatically update the default program for file extensions

### ISSUE THREAD

**Issue by trm217** [cursor/cursor#1052]
I thought I'd try out `Cursor` for a couple of minutes.
I then noticed that all coding related file-extensions now have cursor as their default program.
Don't do that without asking.

**Comment by jakobFNF** [comment]
True. That's annoying.

**Comment by bojdell** [comment]
just noticed the same. perhaps i missed an onboarding selection, but if not it's an unfortunate dark pattern that should be fixed. it's a pain to undo manually

**Comment by bojdell** [comment]
wow, even when you change the default app the logo still remains. huge turn off
<img width="261" alt="image" src="https://github.com/getcursor/cursor/assets/3486165/447764c4-b9fe-4856-8c9a-ae26240dd7a0">

**Comment by truell20** [comment]
> wow, even when you change the default app the logo still remains. huge turn off

Ack, thanks for reporting this. Cursor certainly should never override your default application choices, will be fixed in the next build.

For context, we basically just tell the OS "here's the set of files Cursor can open." On first glance, seems at least a bit tricky to make this configurable in the app code itself with the dependencies we're using, so we may just delete this entirely.

**Comment by ein-bandit** [comment]
major turn off. Please fix this

**Comment by babeal** [comment]
Is this still an issue? It's the reason why I uninstalled the product and why I'm hesitant to reinstall.

**Comment by Danghor** [comment]
Yes, it is still an issue. I installed it recently and it hijacked a lot of file types.

**Comment by babeal** [comment]
Come on guys, this is getting rather frustrating. I would like to try your app. So I figured I would remove vscode and have it reregister itself as the default app during install. I had to delete all vscode settings from the system, but I was finally able to get vscode to update the settings. Then I open my computer this morning and cursor is back as the default. This is not reasonable.

**Comment by babeal** [comment]
Here is a script for those that want to force the extensions back to vscode. After running it, the app icons in Finder might still show cursor. Right click, and select "Get Info" and the correct app icon will show in the Get Info screen and will cause the finder to update. Close and reopen Finder and they should be correct. I'm really hoping, i don't find them back to Cursor in the morning.

I did try with Cursor to write the script itself, but it was incapable.

Install `duti` 
- `brew install duti`

```sh
#!/bin/bash

# Path to lsregister utility
lsregister_path="/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister"
vscode_bundle="com.microsoft.VSCode"

echo "Updating file extensions associated with Cursor to open with Visual Studio Code..."
echo "--------------------------------------------------------------------------------"

# Variables to track the state
is_cursor_bundle=false

# Read the lsregister dump line by line
$lsregister_path -dump | while IFS= read -r line; do
    if [[ "$line" =~ ^bundle: ]]; then
        # Check if the current bundle is Cursor
        if [[ "$line" =~ Cursor ]]; then
            is_cursor_bundle=true
        else
            is_cursor_bundle=false
        fi
    elif $is_cursor_bundle && [[ "$line" =~ ^bindings: ]]; then
        # Extract the extension(s) from the bindings line
        extensions=$(echo "$line" | grep -o "\.[a-z0-9]*")
        for ext in $extensions; do
            echo "Processing extension: $ext"
            duti -s "$vscode_bundle" "${ext#.}" all
        done
        # Reset the state after processing bindings
        is_cursor_bundle=false
    fi
done

killall Finder
killall Dock

echo "Update complete!"
```

**Comment by raquo** [comment]
> Ack, thanks for reporting this. Cursor certainly should never override your default application choices, will be fixed in the next build.

Nov 2023 -> Dec 2024. Still an issue a year later, at least on MacOS. Okay, I can take a hint. You make your business decisions, I'll make mine.

Someone recommended Cursor to me, on Monday I'll go tell them how it screwed up my system.

@babeal 's script works to remove Cursor file associations – thanks a lot!

Unfortunately the file icons are still showing Cursor's app icon, even though I already uninstalled Cursor.

**Comment by konsalex** [comment]
To also remove in MacOS that worked for me:
`/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -u /Applications/Cursor.app`

**Comment by lonesome-rider** [comment]
This is like a very bad joke. There is no easy way to remove these on Windows except doing it one by one or just uninstalling Cursor which is what I did. You might be losing customers for not doing something so basic as this.

**Comment by JeongJuhyeon** [comment]
Major turn off.

**Comment by tharindu-nw** [comment]
Fix this guys. Uninstalling just because of this.

**Comment by mtdrk** [comment]
Ack, still doing this more than a year later.

**Comment by JeongJuhyeon** [comment]
> Fix this guys. Uninstalling just because of this.

FWIW, having tried both, [Cline](https://github.com/cline/cline) is much more competent as an assistant than Cursor compose + It's open source and free + it doesn't pull icky tricks like these, so we're not missing out :)

**Comment by radusalagean** [comment]
Hi @truell20 - paying customer here.
The automatic file association in macOS is bothering me because I am always very intentional with my Cursor usage. When I want to use Cursor, I open it and manually open my project directory, which is already free of any sensitive credentials. However, I don't want to accidentally open a sensitive credentials file with Cursor due to this automatic file association behavio - for [security reasons](https://news.ycombinator.com/item?id=43366751). Manually changing the file association in macOS for every file type away from Cursor is not something that I should have to go through.
Please prioritize this.

**Comment by Truimo** [comment]
This is extremely annoying, and the situation remains the same for now.

**Comment by MajidAskari** [comment]
This is hostile behavior. It immediately breaks workflow and forces users to waste time fixing your unsolicited mess.
@truell20

Moving away from Cursor in our company because of this.

**Comment by DammitJacqui** [comment]
This is incredibly frustrating. Canceling my Cursor account and uninstalling the app because of this. It set .xls, CSV, xcodeproj and other files to open in Cursor by default, and it won't let me revert the setting -- each time I try, it automatically goes back to Cursor as the system default. I wasn't prompted to allow this and it is invasive, hostile and just incredibly off-putting. Why would i EVER want to use Cursor to open Excel files!?

**Comment by leerob** [comment]
Thanks for opening this issue! We've moved our community support to the [Cursor Forum](https://forum.cursor.com) where you can get help from both the team and other users.

If this is a specific bug that you can still reproduce, please open a new issue with clear reproduction steps on the forum.

Thanks you!
